### PR TITLE
changed feed to take into account who user follows

### PR DESCRIPTION
**Pull request by olaokoroafor**
Description:
 - Altered the way Restaurant Server serves restaurants to the users
 - It first displays liked restaurants from by the FoodFriends that the user has followed, and then displays restaurants that people want to go to, then general restaurants in the parse database, and then finally the yelp API

### CODE REVIEW

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java [hunk 1]
```diff
@@ -68,6 +80,147 @@ public void setUser(UserObservable user) {
         this.user = user;
     }
 
+
+    private void populate_friends(){
+        ParseQuery<Friends> query = ParseQuery.getQuery(Friends.class);
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:85]: Could have a FriendsHelper Class

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java [hunk 2]
```diff
@@ -68,6 +80,147 @@ public void setUser(UserObservable user) {
...
+        }
+        else if (source == PARSE_GENERAL){
+            parseGeneral(apiKey, listener);
+        } else {
+            yelpQuery(apiKey, listener);
+        }
+    }
+
+    private void parseLikes(String apiKey, RestaurantListener listener) {
+        ParseQuery<UserLike> query = ParseQuery.getQuery(UserLike.class);
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:117]: Same with a LikesHelper

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java [hunk 3]
```diff
@@ -68,6 +80,147 @@ public void setUser(UserObservable user) {
...
+        // include data referred by restaurant key
+        query.whereContainedIn("user", friends);
+        query.include(UserLike.RESTAURANT_KEY);
+        //TO DO MAKE DISTINCT
+        query.addDescendingOrder("createdAt");
+        query.setLimit(20);
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:123]: Good to make this a `static`

**Review by nammyfb — CHANGES_REQUESTED** [review]

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java (hunk 1 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:85]: Okay!

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java (hunk 2 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:117]: Don't think I'll do this just cause don't want to overdo it, as function is returning restaurants, which makes sense to do in restaurant server

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java (hunk 3 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:123]: Done!

**Pull request merged**